base/uk.ac.stfc.isis.ibex.e4.ui.perspectiveswitcher/src/uk/ac/stfc/isis/ibex/e4/ui/perspectiveswitcher/CopyPerspectiveSnippetProcessor.java

```diff
@@ -1,55 +1,137 @@
 package uk.ac.stfc.isis.ibex.e4.ui.perspectiveswitcher;
 
 import org.eclipse.e4.core.di.annotations.Execute;
+import org.eclipse.e4.core.services.events.IEventBroker;
 import org.eclipse.e4.ui.model.application.MApplication;
+import org.eclipse.e4.ui.model.application.ui.MUIElement;
 import org.eclipse.e4.ui.model.application.ui.advanced.MPerspective;
 import org.eclipse.e4.ui.model.application.ui.advanced.MPerspectiveStack;
+import org.eclipse.e4.ui.model.application.ui.basic.MPartSashContainerElement;
+import org.eclipse.e4.ui.workbench.UIEvents;
+import org.eclipse.e4.ui.workbench.UIEvents.EventTags;
 import org.eclipse.e4.ui.workbench.modeling.EModelService;
 import org.eclipse.e4.ui.workbench.modeling.EPartService;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
 
+import uk.ac.stfc.isis.ibex.e4.ui.perspectiveswitcher.controls.ResetLayoutButtonModel;
 import uk.ac.stfc.isis.ibex.preferences.PreferenceSupplier;
 
 /**
  * Copies all snippet perspectives to perspective stack called
  * "MainPerspectiveStack" In order to register/reset perspective and not have to
  * sync two copies in e4xmi.
- * 
+ *
  */
 public class CopyPerspectiveSnippetProcessor {
 
-    /**
-     * Clone each snippet that is a perspective and add the cloned perspective
-     * to the main PerspectiveStack.
-     * 
-     * @param app
-     *            MApplication
-     * @param partService
-     *            EPartService
-     * @param modelService
-     *            EModelService
-     */
-    @Execute
-    public void execute(MApplication app, EPartService partService, EModelService modelService) {
-        PerspectivesProvider perspectivesProvider = new PerspectivesProvider(app, partService, modelService);
-        MPerspectiveStack perspectiveStack = perspectivesProvider.getTopLevelStack();
-
-        // Only do this when no other children, or the restored workspace state
-        // will be overwritten.
-        if (!perspectiveStack.getChildren().isEmpty()) {
-            return;
-        }
-
-        // clone each snippet that is a perspective and add the cloned
-        // perspective into the main PerspectiveStack
-        boolean isFirst = true;
-        for (MPerspective perspective : perspectivesProvider.getInitialPerspectives()) {	
-        	if (!PreferenceSupplier.perspectivesToHide().contains(perspective.getElementId())) {
-        		perspectiveStack.getChildren().add(perspective);
-                if (isFirst) {
-                    perspectiveStack.setSelectedElement(perspective);
-                    isFirst = false;
-                }
-        	}
-        }
-    }
+	private PerspectivesProvider perspectivesProvider;
```

Comment:
Here you can see that your changes and was there previously are misaligned because the text editor in github does tabs differently than the one in eclipse. If you replace your tabs with spaces it should be aligned.

Reply:
Good to know!